pghoard/basebackup/delta.py

```diff
@@ -75,7 +78,23 @@ def __init__(
         self.data_file_format = data_file_format
 
     def _snapshot(self, snapshotter: Snapshotter) -> SnapshotResult:
-        snapshotter.snapshot(reuse_old_snapshotfiles=False)
+        def progress_callback(msg: ProgressStep, progress_data: ProgressMetrics):
+            key = "snapshot_progress"
+            persisted_progress = PersistedProgress.read(self.metrics)
```

Comment:
This is going to cause a lot of read/writes. We should implement something that mostly updates in memory and writes every now and again as this callback is called a lot AFAIK.

Reply:
will update this